app/src/test/java/tech/relaycorp/courier/data/model/MessageAddressTest.kt

```diff
@@ -21,3 +30,68 @@ internal class MessageAddressTest {
         }
     }
 }
+
+class PublicMessageAddressTest {
+    @Nested
+    inner class Resolve {
+        private val dohClient = mock<DoHClient>()
+
+        private val originalHost = "example.com"
+        private val originalPublicAddress = "https://$originalHost"
+        private val targetHost = "target.example.com"
+        private val targetPort = "1357"
+        private val srvRecordName = "_rcrc._tcp.$originalHost"
+        private val srvRecordData = "0 1 $targetPort $targetHost."
+
+        private val publicAddress = PublicMessageAddress(originalPublicAddress)
+
+        @BeforeEach
+        internal fun setUp() {
+            runBlocking {
+                whenever(dohClient.lookUp(srvRecordName, "SRV")).thenReturn(Answer(listOf(srvRecordData)))
+            }
+        }
+
+        @Test
+        fun `Target host and port should be returned`() = runBlockingTest {
+            val resolvedAddress = publicAddress.resolve(dohClient)
+
+            assertEquals("https://$targetHost:$targetPort", resolvedAddress)
+        }
+
+        @Test
+        fun `Any original scheme, port, path, query string or fragment should be discarded`() = runBlockingTest {
+            val complexPublicAddress =
+                PublicMessageAddress("http://$originalHost:9876/path?query=string#fragment")
+
+            val resolvedAddress = complexPublicAddress.resolve(dohClient)
+
+            assertEquals("https://$targetHost:$targetPort", resolvedAddress)
+        }
```

Comment:
I have a note to replace URLs with domain names in public addresses. That's why the only thing I'm taking is the host name in the URL.

The SRV record already specifies the port, so that's why we're also ignoring any port explicitly specified in the original URL.